### PR TITLE
Drop support for iOS 10

### PR DESCRIPTION
**Pull request by nheagy**
Refs #11683

This PR removes support for iOS 10.
1. Changed targets' deployment target to iOS 11
2. Removed all `available` checks for iOS 11, and their `else` clauses, etc
3. Fixed as many new warnings as I could easily fix

### Areas to test:
It's impossible to anticipate all areas of the app that might be affected but this change, however the following were areas that had warnings in iOS 11 which I fixed:
- PluginDirectoryViewController
- BlogListViewController
- BlogSelectorViewController
- PageListViewController
- AbstractPostListViewController
- ThemeBrowserViewController

### Remaining warnings:
- `'addressDictionary' was deprecated in iOS 11.0: Use @properties` in CLPlacemark+Formatting.swift. Maybe understandable to @SergioEstevao @jleandroperez @frosty ?
- `'automaticallyAdjustsScrollViewInsets' was deprecated in iOS 11.0: Use UIScrollView's contentInsetAdjustmentBehavior instead` in MediaLibraryViewController.swift. Maybe help from @frosty?

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Post/AbstractPostListViewController.swift [hunk 1]
```diff
@@ -349,22 +300,14 @@ class AbstractPostListViewController: UIViewController,
...
 
     fileprivate func configureSearchBackingView() {
         // This mask view is required to cover the area between the top of the search
         // bar and the top of the screen on an iPhone X and on iOS 10.
-        var topAnchor = topLayoutGuide.bottomAnchor
-
-        if #available(iOS 11.0, *) {
-            topAnchor = view.safeAreaLayoutGuide.topAnchor
-        }
+        var topAnchor = view.safeAreaLayoutGuide.topAnchor
```
Comment (frosty) [inline WordPress/Classes/ViewRelated/Post/AbstractPostListViewController.swift]: I get a warning here recommending using a `let`, as `topAnchor` never changes.

**Comment by nheagy** [comment]
> Regarding the Core Location deprecation... looks like we may need to build an address manually using the properties of CLPlacemark

I'm hoping someone that has actually worked on that code is interested in fixing it. I don't even know where it's used 😜

**Comment by nheagy** [comment]
@frosty this is ready for a second review.

**Comment by SergioEstevao** [comment]
> 'addressDictionary' was deprecated in iOS 11.0: Use @properties in CLPlacemark+Formatting.swift. Maybe understandable to @SergioEstevao @jleandroperez @frosty ?

@nheagy Just pushed a commit that sorts out this warning, I hope you don't mind.

**Comment by nheagy** [comment]
I don't mind at all @SergioEstevao, you're my hero!

@frosty I fixed that constraint problem when viewing articles in the reader

**Comment by koke** [comment]
Nice, this would also unlock #9633 🎉

**Comment by nheagy** [comment]
@frosty @shiki thanks for the reviews! 🎉

> I think we can also remove the version check in

Removed 👍

> This is also present in develop.

If there's not an issue for that yet, you should create it!

**Pull request merged**